### ISSUE TITLE
Suddenly no print() output in the console

### ISSUE THREAD

**Issue by ScyDev** [godotengine/godot#1287]
Happens when printing inside a loop. If it's an endless loop or just very many iterations, all the print() calls inside the loop will never show. 
Every print() should be flushed to the output as soon as it is issued.

All you get is:

*\* Debug Process Started **
*\* Debug Process Stopped **

**Comment by reduz** [comment]
yeah prints go via tcp connection and it most likely clogs and disconnects.
Maybe it should have a limit of characters/second and not send more than
that (add something like [too many chars])

**Comment by reduz** [comment]
if you run from commandline debugger (godot -d) you will be able to see the
whole output

**Comment by ScyDev** [comment]
Ok, starting from commandline with -d works.

But still, this is a bug that needs fixing, right?

**Comment by reduz** [comment]
yeah will keep this open until fixed

**Comment by akien-mga** [comment]
Has this been fixed in the current master branch? I think I've already seen a "too many chars" output when printing a lot of stuff, but I'd have to check.

**Comment by bojidar-bg** [comment]
I think this is fixed, since I had the same problem once, but no more :smile: Still, better to wait for confirmation from @ScyDev or @reduz .

**Comment by akien-mga** [comment]
Closing as fixed, will reopen if it's not the case.

**Comment by OdellDotson** [comment]
Still getting this bug on the OS X version...

**Comment by LordHexahedron** [comment]
Not getting any print() output under windows at all, current steam version of godot so I'd wager it's reasonably up to date.

**Comment by mundusnine** [comment]
I have the same issue on linux version of Godot 2.1.3-stable

**Comment by blurymind** [comment]
running 2.1.4rc - I have this bug

**Comment by bojidar-bg** [comment]
Ah, actually, rereading the OP's post:
> If it's an endless loop or just very many iterations, all the print() calls inside the loop will never show.

I don't think that's fixed yet.

**Comment by blurymind** [comment]
running 2.1.4 

Try this:
```
func _fixed_process(delta):
	print("this wont print in godot editor, but will print in cdm line, regression from 2.1.2 ")
```

**Comment by akien-mga** [comment]
> Changing "Editor Setting -> Network Host -> Debug Host" to localhost(127.0.0.1) solved the issue in 2.1.4 stable. (Windows 10 x64)

CC @Faless.

**Comment by Faless** [comment]
Yeah, I was suspecting this actually, apparently link-local ipv6 addresses (i.e. `fe80::/64`) does not work. Not sure why. I think we could just remove them as an option.
Also, the editor should pick `127.0.0.1` by default (assuming no one disables ipv4) and not the first one returned by the OS.
I'll work on a PR and we can discuss it further there

**Comment by groud** [comment]
Is that one fixed then ?

**Comment by odragora** [comment]
> Is that one fixed then ?
No. I had nothing in editor output console.

> Changing "Editor Setting -> Network Host -> Debug Host" to localhost(127.0.0.1) solved the issue in 2.1.4 stable.
That worked for me.
Windows 10 x64, Godot 2.1.4 stable.

**Comment by odragora** [comment]
> Is that one fixed then ?

No. I had nothing in editor output console.
There were "Remote Debugger: Connection failed with status '1'" messages in Godot console window instead.

> Changing "Editor Setting -> Network Host -> Debug Host" to localhost(127.0.0.1) solved the issue in 2.1.4 stable.

That worked for me.
Windows 10 x64, Godot 2.1.4 stable.

**Comment by sicienss** [comment]
**PROBLEM:**

Suddenly I can no longer print to the console in Godot 3.0.2 stable.

I can print in a new project, but not my old one.

Even when I delete all files and singletons in my project except for a simple node with a script 

```
func _ready():
    print("test")
```

I get no output except the normal "Debugger Process Started" message. I don't even see the normal "OpenGL ES 3.0 Renderer..." message.

I have seen a few questions online about this problem for Godot 2.0, and the answers involve changing the debug_port from project settings to 127.0.0.1. My port is already listed as 127.0.0.1, with remote port 6007.

**OTHER INFO:**
I'm on MacOS Sierra 10.12.6.

I don't think this is an overflow issue since I am trying to print a single line.

When I start Godot, I click to allow Godot to accept incoming network connections. I also checked that Godot can accept connections from my Privacy Settings.

This may not be related, but when I run my code, I get the following warning:

> Remote Debugger: Connection failed with status: '2', retrying in 1 msec.

The debugger seems to be working in that it still kills the run on errors.
